### PR TITLE
openstack-crowbar: refactor top level playbooks

### PR DESCRIPTION
**Pull request by stefannica**
Refactor the ansible playbooks and roles such that that every
stage in the Jenkinsfile corresponds to one and only one playbook,
instead of having to call the `deploy-crowbar.yml` playbook multiple
times from the same Jenkinsfile stage.

This change is required to further decouple the back-end pipeline
implementation from the Jenkinsfile definition and also in preparation
of follow-up stages to come (e.g. SES integration), that will require
adding even more back-end logic to these stages.

The following playbooks and roles are added and/or updated by
this refactoring change:
 - `crowbar_setup` role (replaces the previous `crowbar_deploy`
 role) - a simple task wrapping a `qa_crowbarsetup.sh`
 command being executed on the admin node. It is called by the
 following top-level Crowbar playbooks, sometime even multiple times
 from the same playbook, to run relevant `qa_crowbarsetup.sh` commands.
 - `install-crowbar.yml` playbook - installs Crowbar on the admin node.
 - `register-crowbar-nodes.yml` - registers crowbar nodes, same as before
 - `deploy-crowbar.yml` playbook (used to be just a wrapper for the
 `crowbar_deploy` role) - now this is the top-level playbook that deploys
 the Crowbar batch scenario
 - `run-crowbar-tests.yml` - runs tests on the deployed Crowbar cloud.

This PR also includes a couple of bug fixes required for the `crowbar_register`
ansible role and support for configuring additional repositories via the `extra_repos`
parameter.

### CODE REVIEW

File: jenkins/ci.suse.de/pipelines/openstack-crowbar.Jenkinsfile [hunk 1]
```diff
@@ -154,9 +154,7 @@ pipeline {
            // This step does the following on the admin node:
            //  - sets up SLES and Cloud repositories
            //  - installs crowbar
-           ardana_lib.ansible_playbook('deploy-crowbar', '-e \'qa_crowbarsetup_cmd="onadmin_runlist prepareinstallcrowbar"\'')
-           ardana_lib.ansible_playbook('deploy-crowbar', '-e \'qa_crowbarsetup_cmd="onadmin_runlist bootstrapcrowbar"\'')
-           ardana_lib.ansible_playbook('deploy-crowbar', '-e \'qa_crowbarsetup_cmd="onadmin_runlist installcrowbar"\'')
+           ardana_lib.ansible_playbook('install-crowbar')
```
Comment (rsalevsky) [inline jenkins/ci.suse.de/pipelines/openstack-crowbar.Jenkinsfile]: Can we rename this lib? It's a bit confusing to see a ardana lib in crowbar code.

File: scripts/jenkins/ardana/ansible/install-crowbar.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
+---
```
Comment (rsalevsky) [inline scripts/jenkins/ardana/ansible/install-crowbar.yml]: Again something similar why is a crowbar file in a ardana related folder?

File: jenkins/ci.suse.de/pipelines/openstack-crowbar.Jenkinsfile (hunk 1 above)
Comment (stefannica) [reply jenkins/ci.suse.de/pipelines/openstack-crowbar.Jenkinsfile]: Yeah, I know, it's really annoying having Crowbar code labeled Ardana, and it's more extensive than you can imagine. It's not really part of this PR, but we do have a Jira card tracking this type of refactoring work (https://jira.suse.de/browse/SCRD-8475 ).

File: scripts/jenkins/ardana/ansible/install-crowbar.yml (hunk 2 above)
Comment (stefannica) [reply scripts/jenkins/ardana/ansible/install-crowbar.yml]: As I was saying in the previous comment, the Crowbar ECP CI is based heavily on the Ardana baseline (ansible playbooks). We need to do this refactoring work, but it's not on our list of priorities (see https://jira.suse.de/browse/SCRD-8475)

**Comment by stefannica** [comment]
> LGTM. Compared to older implementations, I appreciate the move out of jenkins pipeline and its arguments and into ansible, it lowers the entry barrier (and it's more reusable).

Thank you, yes, that was my intention.

**Pull request merged**